src/tools/perf/perftest_run.c

```diff
@@ -262,10 +273,30 @@ static ucs_status_t read_batch_file(FILE *batch_file, const char *file_name,
     return UCS_OK;
 }
 
+static size_t get_test_count(FILE *batch_file)
```

Comment:
instead of doing this, maybe we could send a FIN message to the daemon to exit, after reaching the end of the batch file? that means we would send most FIN messages after getting the next test from the batch file (instead of immediately after running the test), and the last FIN will be outside of the loop

Reply:
I see what you mean, but I think this would not fly because it is perf test who owns EP-to-daemon. Individual perf test is self-contained, in the sense that on startup it initializes the EP to daemon, and at the end of the test it destroys that EP. So there is no mean to send FIN message over this EP afterwards, after the test cleanup is executed.

But in principle I agree that daemon is a kind of shared resource, and ideally it should be "owned" not by an individual test, but rather by a test runner. However this means that we need to have EP-to-daemon outside of the perf-test, somewhere in perftest_run.c:
```
perftest_run:run_test
    run_test_recurs
        ucx_perf_run
            ucp_perf_setup: create EP2D, -> [INIT]
            ucp_perf_cleanup: -> [FIN keep-running], destroy EP2D // never kill daemon from test, as it's shared resource
        ucx_perf_run ...
        ucx_perf_run ...
    kill_daemon: create EP2D, -> [FIN kill], destroy EP2D
```
That's the cleanest option IMO, what do you think?

Reply:
I think it's a good idea!

Reply:
done